app/src/pages/inside/projectSettingsPageContainer/content/integrations/integrations.jsx

```diff
@@ -56,6 +56,11 @@ export const Integrations = () => {
     setPlugin({});
   }, []);
 
+  const goToInitialPage = () => {
+    setPlugin({});
+    dispatch(redirect(initialPage));
```

Comment:
You can add a `prevPageSelector` to extract `type`, `query` and `payload` from the `location` state, because user need to be redirected to the previous page, not to the initial.

![image](https://user-images.githubusercontent.com/31861805/214270350-466d5df0-8b93-41c5-8994-9904a4601b33.png)

And then just redirect to it.